### PR TITLE
Add Foundation Classes for New Architecture (Plan 1)

### PR DESCRIPTION
**Pull request by shukebeta**
## Summary

This PR implements the core foundation classes for our new architecture, completing **Plan 1** of the architecture improvement strategy. These classes provide standardized patterns for API and Service layers without any breaking changes to existing functionality.

### 🏗️ Core Classes
- **`ApiResponseV2<T>`** - Unified response wrapper with success/error states
- **`BaseApi`** - Common HTTP operations with standardized error handling  
- **`BaseService`** - Business logic patterns with validation utilities
- **`ServiceException` hierarchy** - Comprehensive error type system

### 🚨 Exception Types
- `NetworkException` - Connection, timeout, network-level errors
- `ApiBusinessException` - Business logic errors from backend
- `AuthenticationException` - Auth/authorization errors (401/403)
- `DataException` - Validation and parsing errors
- `ServerException` - Server-side errors (5xx)

### ✨ Key Features
- Type-safe response processing with generics
- Automatic error conversion from Dio exceptions
- Input validation helpers (string, numeric, pagination)
- Protected methods for extensibility
- Comprehensive error context and messaging

## Testing Coverage

- **100+ unit tests** across all foundation classes
- Edge case handling for null data, network errors, parsing failures
- Type safety validation and error boundary testing
- Integration scenarios for real-world usage patterns
- All tests passing ✅

## Impact & Verification

- ✅ **Zero Breaking Changes** - Existing app functionality verified identical
- ✅ **No Dependencies** - Foundation classes coexist with current code
- ✅ **Ready for Migration** - Solid base for upcoming API+Service refactoring
- ✅ **Well Documented** - Comprehensive inline documentation
- ✅ **Type Safe** - Full generic support with proper nullable handling

## Next Steps

This foundation enables:
1. **Plan 2** - Constants and configuration cleanup
2. **Plan 3+** - Incremental API+Service pair migrations
3. **Future Development** - Consistent patterns for new features

## Files Added

```
lib/common/foundation/
├── api_response_v2.dart        # Unified response wrapper
├── service_exceptions.dart     # Exception hierarchy  
├── base_api.dart              # HTTP operations base class
├── base_service.dart          # Business logic base class
└── foundation.dart            # Barrel export file

test/common/foundation/
├── api_response_v2_test.dart
├── service_exceptions_test.dart
├── base_api_test.dart
└── base_service_test.dart
```

## Test Plan

- [x] All foundation unit tests pass
- [x] Static analysis clean (flutter analyze)
- [x] Existing app functionality verified identical
- [x] No breaking changes to current API/Service usage

Ready for review\! 🚀

### CODE REVIEW

File: lib/common/foundation/api_response_v2.dart [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+  final T? data;
+  
+  /// Whether the API call was successful
+  final bool isSuccess;
+  
+  /// Error message if the call failed
+  final String? errorMessage;
+  
+  /// HTTP status code from the response
+  final int? statusCode;
```
Comment (shukebeta) [inline lib/common/foundation/api_response_v2.dart:16]: As our server end won't provide statusCode as part of the response, we can safely remove it from ApiResponseV2
